DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/component_interfaces_comparison/pipes/Readme.md

```diff
@@ -0,0 +1,184 @@
+# Streaming Data Interfaces (Pipes)
+This implementation uses a register-mapped invocation interface, and demonstrates how to use pipes to configure a streaming data interface.
+
+![](../assets/agent_n_pipe.svg)
+
+## Invocation Interface
+By default, an un-decorated oneAPI kernel will have all its control signals and arguments mapped into the IP component's control/status register (CSR).
+
+## Data Interface - Streaming (Pipes)
+Pipes are a first-in first-out (FIFO) buffer construct that provide links between elements of a design. Pipes that connect a host and a device are referred to as host pipes. In this design, interfaces `InputPipeA`, `InputPipeB`, `OutputPipeC` are implemented as host pipes.
+
+### Declare Pipe
+Each individual pipe is a program scope class declaration of the templated pipe class:
+```cpp
+sycl::ext::intel::experimental::pipe<id, type, min_capacity, properties>;
+```
+Detailed explanation for each parameters and a list of properties can be found in this dedicated [host pipe](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/Features/experimental/hostpipes) code sample.
+
+### Example of Pipe declearation
+```cpp
+using PipeProps = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::ready_latency<0>));
+class ID_PipeA;
+using InputPipeA =
+    sycl::ext::intel::experimental::pipe<ID_PipeA, int, 0, PipeProps>;
+```
+
+### Pipe API
+Pipes expose read and write interfaces that allow a single element to be read or written in FIFO order to the pipe.
+
+Go to [host pipe](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/C%2B%2BSYCL_FPGA/Tutorials/Features/experimental/hostpipes) code sample for detailed read/write API.
+
+> **Note:** \
```

Comment:
I don't think we need to include these notes here since we are already linking to the other host pipes sample.

In general, I think we would want to either repeat the information in full (so the IPA samples are independent of the FS samples), or, refer users to the FS sample for all common information they share. That way for a given user, there is always just one source they should be referring to for each piece of information.